src/lewis/adapters/stream.py

```diff
@@ -633,12 +668,18 @@ class StreamInterface(InterfaceBase):
 
     Many hardware devices use a protocol that is based on exchanging text with a client via
     a TCP stream. Sometimes RS232-based devices are also exposed this way via an adapter-box.
-    This adapter makes it easy to mimic such a protocol, in a subclass only three members must
-    be overridden:
+    This adapter makes it easy to mimic such a protocol.
+
+    This class has the following attributes which may be overridden by subclasses:
 
+     - protocol: What this interface is called for purposes of the -p commandline option.
+       Defaults to "stream".
      - in_terminator, out_terminator: These define how lines are terminated when transferred
        to and from the device respectively. They are stripped/added automatically.
-       The default is ``\\r``.
+       Inverse of protocol file InTerminator and OutTerminator. The default is ``\\r``.
+     - readtimeout: How many msec to wait for additional data between packets, once transmission
```

Comment:
I remember having the same thought about `None` and looking at asynchat code and then deciding to use empty string instead... but I can't quite remember why exactly. Looking at asynchat again, `None` should be fine (`set_terminator` docstring even says so). I think it was other reasons that made me lean towards `''` over `None`:

An empty string is also what the protocol files use (at least the one Sam showed me that triggered this change). And, also, an empty string could not mean anything different than "no terminator". So I don't think there's anything wrong with recommending it, or any reason to require None instead.